packages/ember-data/lib/adapters/rest-adapter.js

```diff
@@ -412,14 +412,14 @@ var RESTAdapter =  Adapter.extend(BuildURLMixin, {
     @param {Object} query
     @return {Promise} promise
   */
-  query: function(store, type, query) {
-    var url = this.buildURL(type.modelName, null, null, 'query', query);
+  query: function(store, type, options) {
+    var url = this.buildURL(type.modelName, null, null, 'query', options.query);
```

Comment:
The changes to `buildURL` above make it seem like `options` should be passed in here...?